vm-rust/src/player/handlers/datum_handlers/script.rs

```diff
@@ -39,6 +39,7 @@ impl ScriptDatumHandlers {
 
   pub fn create_script_instance(script_ref: &CastMemberRef) -> (ScriptInstanceRef, DatumRef) {
     reserve_player_mut(|player| {
+      // TODO we are allocating two script instance ids here, should only be one
```

Comment:
Are you referring to `get_free_script_instance_id`? It doesn't allocate an instance id, only returns the next free id.

Reply:
Yes - to be more precise we're calling `get_free_script_instance_id` twice: once here and then inside `alloc_script_instance`. I assume we only want to call it once per script instance creation.

Reply:
Thinking I should I change `alloc_script_instance` to use the `id` field of ScriptInstance or explicitly accept an id argument.